# Analysis for the process of 5 Questions for 5 Agent Chats                   CLUSTER (START)

In [1]:
import pandas as pd
import numpy as np
import seaborn as sn
import scipy.stats as st
import statsmodels.stats.multicomp as ml
from scipy.stats import f_oneway
from statsmodels.stats.multicomp import pairwise_tukeyhsd
import scikit_posthocs as sp
import matplotlib
import matplotlib.pyplot as plt 
import scipy.stats as stats
%matplotlib inline

In [2]:
#ΕΙΣΑΓΩΓΗ ΔΕΔΟΜΕΝΩΝ
df = pd.read_excel('MOOC3.xlsx', sheet_name= "GENERAL", skiprows=1)

In [3]:
#ΕΞΑΓΩΓΗ ΤΩΝ ΣΤΗΛΩΝ ΜΕ ΤΑ ΕΡΩΤΗΜΑΤΑ ΚΑΙ Μ.Ο ΒΑΥΜΟΛΟΓΙΑΣ ΚΑΘΕ ΦΟΙΤΗΤΗ

dfinfo=df[[ 'Cluster (START)',  'Subgroup']]
dfbot1a=df[df.columns[127:128:1]]
dfbot1b=df[df.columns[129:133:1]]

dfbot2a=df[df.columns[135:136:1]]
dfbot2b=df[df.columns[137:141:1]]

dfbot3a=df[df.columns[143:144:1]]
dfbot3b=df[df.columns[145:149:1]]

dfbot4a=df[df.columns[151:152:1]]
dfbot4b=df[df.columns[153:157:1]]

dfbot5a=df[df.columns[159:160:1]]
dfbot5b=df[df.columns[161:165:1]]


result = [dfinfo, dfbot1a,dfbot1b, dfbot2a, dfbot2b, dfbot3a, dfbot3b, dfbot4a, dfbot4b, dfbot5a, dfbot5b]

  
dfbot_total= pd.concat(result, axis=1)
dfbot_total

Cluster (START)  Subgroup  \
0                 2.0       1.0   
1                 2.0       1.0   
2                 NaN       NaN   
3                 NaN       NaN   
4                 2.0       1.0   
...               ...       ...   
2655              NaN       NaN   
2656              2.0       1.0   
2657              2.0       1.0   
2658              NaN       NaN   
2659              NaN       NaN   

     Στη συζήτηση που είχα στο chat νομίζω ότι συνεργάστηκα καλά με τον/την συνεργάτη μου.  \
0                                                   NaN                                      
1                                                   NaN                                      
2                                                   NaN                                      
3                                                   NaN                                      
4                                                     4                                      
...                                                 ...                                      
2655                                                NaN                                      
2656                                                NaN                                      
2657                                                  5                                      
2658                                                NaN                                      
2659                                                NaN                                      

     Μού άρεσε που υπήρχε ο πράκτορας και έκανε παρεμβάσεις στη συζήτηση.  \
0                                                   NaN                     
1                                                   NaN                     
2                                                   NaN                     
3                                                   NaN                     
4                                                     5                     
...                                                 ...                     
2655                                                NaN                     
2656                                                NaN                     
2657                                                  3                     
2658                                                NaN                     
2659                                                NaN                     

     Νομίζω ότι οι παρεμβάσεις του πράκτορα ήταν σχετικές με το τι συζητούσαμε εκείνη τη στιγμή.  \
0                                                   NaN                                            
1                                                   NaN                                            
2                                                   NaN                                            
3                                                   NaN                                            
4                                                     5                                            
...                                                 ...                                            
2655                                                NaN                                            
2656                                                NaN                                            
2657                                                  3                                            
2658                                                NaN                                            
2659                                                NaN                                            

     Νομίζω ότι οι παρεμβάσεις του πράκτορα με έκαναν να σκεφτώ καλύτερα γι αυτό που μαθαίνω.  \
0                                                   NaN                                         
1                                                   NaN                                         
2                                      

In [4]:
#EΛΕΓΧΟΣ ΓΙΑ ΚΑΘΕ ΜΑΘΗΤΗ ΑΝ ΕΧΕΙ ΑΠΑΝΤΗΣΕΙ ΣΕ ΤΟΥΛΑΧΙΣΤΟΝ 2 CHAT ΣΥΝΟΛΙΚΑ. ΑΝ ΕΧΕΙ <2 AΠΑΝΤΗΜΕΝΑ CHAT ΤΟ ΚΑΝΩ DROP 

dfbot_total.replace(to_replace=float('nan'),value='-', inplace=True)
for i in range(dfbot_total.shape[0]):
    counter = 0
    if dfbot_total.at[i,'Subgroup'] == 1:
        
        for j in range(2,27,1):
            
            if dfbot_total.at[i,dfbot_total.columns[j]]!="-":             
                counter+=1                       
        if counter<=10:
            
            dfbot_total.drop(df.index[i], inplace=True)           
    else:
        dfbot_total.drop(df.index[i], inplace=True)
        
          
dfbot_total.replace(to_replace='-',value=float('nan'), inplace=True)
dfbot_total

Cluster (START)  Subgroup  \
4                 2.0       1.0   
19                3.0       1.0   
25                2.0       1.0   
28                1.0       1.0   
38                3.0       1.0   
...               ...       ...   
2620              3.0       1.0   
2642              4.0       1.0   
2647              3.0       1.0   
2652              3.0       1.0   
2657              2.0       1.0   

     Στη συζήτηση που είχα στο chat νομίζω ότι συνεργάστηκα καλά με τον/την συνεργάτη μου.  \
4                                                     4                                      
19                                                    4                                      
25                                                    3                                      
28                                                    1                                      
38                                                    5                                      
...                                                 ...                                      
2620                                                  5                                      
2642                                                  5                                      
2647                                                  5                                      
2652                                                  5                                      
2657                                                  5                                      

     Μού άρεσε που υπήρχε ο πράκτορας και έκανε παρεμβάσεις στη συζήτηση.  \
4                                                     5                     
19                                                    2                     
25                                                    4                     
28                                                    1                     
38                                                    5                     
...                                                 ...                     
2620                                                  1                     
2642                                                  4                     
2647                                                  5                     
2652                                                  3                     
2657                                                  3                     

     Νομίζω ότι οι παρεμβάσεις του πράκτορα ήταν σχετικές με το τι συζητούσαμε εκείνη τη στιγμή.  \
4                                                     5                                            
19                                                    4                                            
25                                                    4                                            
28                                                    1                                            
38                                                    5                                            
...                                                 ...                                            
2620                                                  2                                            
2642                                                  4                                            
2647                                                  5                                            
2652                                                  3                                            
2657                                                  3                                            

     Νομίζω ότι οι παρεμβάσεις του πράκτορα με έκαναν να σκεφτώ καλύτερα γι αυτό που μαθαίνω.  \
4                                                     4                                         
19                                                    4                                         
25                                     

In [5]:
#ΔΗΜΙΟΥΡΓΙΑ 5 ΝΕΑ DF ΜΕ ΤΗΝ ΚΑΘΕ ΕΡΩΤΗΣΗ ΚΑΙ ΓΙΑ ΤΙΣ 5 ΕΒΔΟΜΑΔΕΣ ΣΥΝΟΛΙΚΟ ΠΛΗΘΟΣ 523

#Στη συζήτηση που είχα στο chat νομίζω ότι συνεργάστηκα καλά με τον/την συνεργάτη μου.
dfinfo=dfbot_total[[ 'Cluster (START)',  'Subgroup']]
dfchat1a=dfbot_total[dfbot_total.columns[2:3:1]]
dfchat1b=dfbot_total[dfbot_total.columns[3:4:1]]
dfchat1c=dfbot_total[dfbot_total.columns[4:5:1]]
dfchat1d=dfbot_total[dfbot_total.columns[5:6:1]]
dfchat1e=dfbot_total[dfbot_total.columns[6:7:1]]

#Μού άρεσε που υπήρχε ο πράκτορας και έκανε παρεμβάσεις στη συζήτηση.
dfchat2a=dfbot_total[dfbot_total.columns[7:8:1]]
dfchat2b=dfbot_total[dfbot_total.columns[8:9:1]]
dfchat2c=dfbot_total[dfbot_total.columns[9:10:1]]
dfchat2d=dfbot_total[dfbot_total.columns[10:11:1]]
dfchat2e=dfbot_total[dfbot_total.columns[11:12:1]]

#Νομίζω ότι οι παρεμβάσεις του πράκτορα ήταν σχετικές με το τι συζητούσαμε εκείνη τη στιγμή.
dfchat3a=dfbot_total[dfbot_total.columns[12:13:1]]
dfchat3b=dfbot_total[dfbot_total.columns[13:14:1]]
dfchat3c=dfbot_total[dfbot_total.columns[14:15:1]]
dfchat3d=dfbot_total[dfbot_total.columns[15:16:1]]
dfchat3e=dfbot_total[dfbot_total.columns[16:17:1]]

#Νομίζω ότι οι παρεμβάσεις του πράκτορα με έκαναν να σκεφτώ καλύτερα γι αυτό που μαθαίνω.
dfchat4a=dfbot_total[dfbot_total.columns[17:18:1]]
dfchat4b=dfbot_total[dfbot_total.columns[18:19:1]]
dfchat4c=dfbot_total[dfbot_total.columns[19:20:1]]
dfchat4d=dfbot_total[dfbot_total.columns[20:21:1]]
dfchat4e=dfbot_total[dfbot_total.columns[21:22:1]]

#Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου.
dfchat5a=dfbot_total[dfbot_total.columns[22:23:1]]
dfchat5b=dfbot_total[dfbot_total.columns[23:24:1]]
dfchat5c=dfbot_total[dfbot_total.columns[24:25:1]]
dfchat5d=dfbot_total[dfbot_total.columns[25:26:1]]
dfchat5e=dfbot_total[dfbot_total.columns[26:27:1]]


result1 = [dfchat1a, dfchat2a, dfchat3a, dfchat4a, dfchat5a]
dfchat1= pd.concat(result1, axis=1)
dfchat1


result2 = [dfchat1b, dfchat2b, dfchat3b, dfchat4b, dfchat5b]
dfchat2= pd.concat(result2, axis=1)
dfchat2


result3 = [dfchat1c, dfchat2c, dfchat3c, dfchat4c, dfchat5c]
dfchat3= pd.concat(result3, axis=1)
dfchat3


result4 = [dfchat1d, dfchat2d, dfchat3d, dfchat4d, dfchat5d]
dfchat4= pd.concat(result4, axis=1)
dfchat4


result5 = [dfchat1e, dfchat2e, dfchat3e, dfchat4e, dfchat5e]
dfchat5= pd.concat(result5, axis=1)
dfchat5


Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου.  \
4                                                     1                                   
19                                                    3                                   
25                                                    1                                   
28                                                    1                                   
38                                                    1                                   
...                                                 ...                                   
2620                                                  5                                   
2642                                                  3                                   
2647                                                  2                                   
2652                                                  3                                   
2657                                                  3                                   

     Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου..1  \
4                                                     1                                     
19                                                    3                                     
25                                                    3                                     
28                                                    2                                     
38                                                    1                                     
...                                                 ...                                     
2620                                                  5                                     
2642                                                  3                                     
2647                                                  1                                     
2652                                                  3                                     
2657                                                  3                                     

      Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου..2  \
4                                                   NaN                                      
19                                                  2.0                                      
25                                                  1.0                                      
28                                                  2.0                                      
38                                                  1.0                                      
...                                                 ...                                      
2620                                                5.0                                      
2642                                                3.0                                      
2647                                                4.0                                      
2652                                                3.0                                      
2657                                                3.0                                      

      Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου..3  \
4                                                   NaN                                      
19                                                  2.0                                      
25                                                  1.0                                      
28                                                  1.0                                      
38                                                  1.0                                      
...                                                 ...                                      
2620    

In [6]:
#ΜΕΣΟΣ ΟΡΟΣ ΚΑΘΕ ΕΡΩΤΗΣΗΣ ΚΑΙ ΣΤΙΣ 5 ΕΒΔΟΜΑΔΕΣ ΤΟΥ ΚΑΘΕ ΦΟΙΤΗΤΗ

dfchat1['mean'] = dfchat1.mean(axis=1, skipna=True)
res1= [dfinfo, dfchat1['mean']]
dfquest1= pd.concat(res1, axis=1)
dfquest1

dfchat2['mean'] = dfchat2.mean(axis=1, skipna=True)
res2= [dfinfo, dfchat2['mean']]
dfquest2= pd.concat(res2, axis=1)
dfquest2

dfchat3['mean'] = dfchat3.mean(axis=1, skipna=True)
res3= [dfinfo, dfchat3['mean']]
dfquest3= pd.concat(res3, axis=1)
dfquest3

dfchat4['mean'] = dfchat4.mean(axis=1, skipna=True)
res4= [dfinfo, dfchat4['mean']]
dfquest4= pd.concat(res4, axis=1)
dfquest4

dfchat5['mean'] = dfchat5.mean(axis=1, skipna=True)
res5= [dfinfo, dfchat5['mean']]
dfquest5= pd.concat(res5, axis=1)
dfquest5

Cluster (START)  Subgroup      mean
4                 2.0       1.0  1.000000
19                3.0       1.0  2.000000
25                2.0       1.0  1.333333
28                1.0       1.0  1.666667
38                3.0       1.0  1.000000
...               ...       ...       ...
2620              3.0       1.0  5.000000
2642              4.0       1.0  3.000000
2647              3.0       1.0  3.000000
2652              3.0       1.0  3.333333
2657              2.0       1.0  3.000000

[523 rows x 3 columns]

# CLUSTER(START) QUESTION 1 
#Στη συζήτηση που είχα στο chat νομίζω ότι συνεργάστηκα καλά με τον/την συνεργάτη μου.

In [10]:
#Cluster Start 1 (LOW>4)
q1_1=dfquest1.groupby('Cluster (START)').get_group(1)


col=q1_1.columns
q1low=q1_1[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q1low.dropna(inplace=True)

meanq1low = q1low.mean(axis=0)#μεσος ορος μαθητων low στην ερώτηση 1


q1low_list=[]
for i, row in q1low.iterrows():
    q1low_list.append(q1low.at[i,'mean'])


q1low_list


<ipython-input-10-6388357fdcaf>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q1low.dropna(inplace=True)


[4.0,
 4.333333333333333,
 3.0,
 1.0,
 3.6666666666666665,
 2.0,
 3.0,
 5.0,
 3.6666666666666665,
 1.6666666666666667,
 3.6666666666666665,
 4.0,
 3.3333333333333335,
 3.0,
 5.0,
 1.6666666666666667,
 3.6666666666666665,
 3.6666666666666665,
 4.333333333333333,
 2.6666666666666665,
 5.0,
 2.6666666666666665,
 2.0,
 5.0,
 5.0,
 3.6666666666666665,
 4.0]

In [11]:
#Cluster Start 2 (MEDIUM 4-5)
q1_2=dfquest1.groupby('Cluster (START)').get_group(2)


col=q1_2.columns
q1medium=q1_2[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q1medium.dropna(inplace=True)

meanq1medium = q1medium.mean(axis=0)#μεσος ορος μαθητων Medium στην ερώτηση 1


q1medium_list=[]
for i, row in q1medium.iterrows():
    q1medium_list.append(q1medium.at[i,'mean'])


q1medium_list


<ipython-input-11-f3723a02a1bf>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q1medium.dropna(inplace=True)


[5.0,
 4.333333333333333,
 3.0,
 5.0,
 4.0,
 5.0,
 3.6666666666666665,
 3.3333333333333335,
 5.0,
 2.6666666666666665,
 2.0,
 3.3333333333333335,
 5.0,
 5.0,
 4.666666666666667,
 5.0,
 3.6666666666666665,
 4.666666666666667,
 3.6666666666666665,
 5.0,
 4.333333333333333,
 4.0,
 3.6666666666666665,
 4.666666666666667,
 5.0,
 5.0,
 3.0,
 3.0,
 4.5,
 3.6666666666666665,
 5.0,
 5.0,
 4.333333333333333,
 5.0,
 5.0,
 4.5,
 4.0,
 4.333333333333333,
 5.0,
 4.666666666666667,
 4.666666666666667,
 5.0,
 4.333333333333333,
 5.0,
 4.0,
 5.0,
 3.0,
 4.0,
 3.6666666666666665,
 5.0,
 4.666666666666667,
 2.3333333333333335,
 5.0,
 4.333333333333333,
 5.0,
 3.0,
 3.6666666666666665,
 3.6666666666666665,
 5.0,
 5.0,
 3.5,
 3.3333333333333335,
 4.666666666666667,
 2.3333333333333335,
 4.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 4.0,
 3.5,
 3.5,
 4.0,
 4.0,
 4.333333333333333,
 4.0,
 3.0,
 5.0,
 5.0,
 4.666666666666667,
 5.0,
 3.6666666666666665,
 4.666666666666667,
 5.0,
 4.666666666666667,
 3.6666666666666665,


In [12]:
#Cluster Start 3 (LARGE 5-6)
q1_3=dfquest1.groupby('Cluster (START)').get_group(3)


col=q1_3.columns
q1large=q1_3[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q1large.dropna(inplace=True)

meanq1large = q1large.mean(axis=0)#μεσος ορος μαθητων Large στην ερώτηση 1


q1large_list=[]
for i, row in q1large.iterrows():
    q1large_list.append(q1large.at[i,'mean'])


q1large_list


<ipython-input-12-dcc1ae0b5dee>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q1large.dropna(inplace=True)


[4.333333333333333,
 5.0,
 5.0,
 4.333333333333333,
 5.0,
 5.0,
 3.6666666666666665,
 2.0,
 4.333333333333333,
 5.0,
 5.0,
 4.0,
 5.0,
 2.6666666666666665,
 4.0,
 4.0,
 4.0,
 5.0,
 5.0,
 2.3333333333333335,
 5.0,
 4.666666666666667,
 4.666666666666667,
 5.0,
 5.0,
 3.6666666666666665,
 3.0,
 1.6666666666666667,
 4.333333333333333,
 5.0,
 5.0,
 4.0,
 4.5,
 5.0,
 4.0,
 5.0,
 3.6666666666666665,
 5.0,
 4.0,
 5.0,
 3.6666666666666665,
 5.0,
 4.666666666666667,
 4.666666666666667,
 5.0,
 5.0,
 5.0,
 3.6666666666666665,
 5.0,
 3.3333333333333335,
 4.0,
 4.5,
 4.666666666666667,
 5.0,
 4.0,
 4.666666666666667,
 5.0,
 1.0,
 5.0,
 5.0,
 2.3333333333333335,
 5.0,
 5.0,
 5.0,
 4.0,
 5.0,
 4.0,
 5.0,
 3.0,
 5.0,
 4.666666666666667,
 4.666666666666667,
 5.0,
 5.0,
 3.6666666666666665,
 4.5,
 5.0,
 4.666666666666667,
 4.0,
 4.0,
 5.0,
 5.0,
 4.0,
 5.0,
 4.0,
 3.6666666666666665,
 4.666666666666667,
 4.666666666666667,
 4.666666666666667,
 4.666666666666667,
 5.0,
 3.6666666666666665,
 4.666666666666

In [13]:
#Cluster Start 4 (XLARGE <6)
q1_4=dfquest1.groupby('Cluster (START)').get_group(4)


col=q1_4.columns
q1xlarge=q1_4[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q1xlarge.dropna(inplace=True)

meanq1xlarge = q1xlarge.mean(axis=0)#μεσος ορος μαθητων XLarge στην ερώτηση 1


q1xlarge_list=[]
for i, row in q1xlarge.iterrows():
    q1xlarge_list.append(q1xlarge.at[i,'mean'])


q1xlarge_list

<ipython-input-13-9d9e7b80f747>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q1xlarge.dropna(inplace=True)


[5.0,
 4.333333333333333,
 5.0,
 3.0,
 4.666666666666667,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 5.0,
 4.0,
 4.333333333333333,
 5.0,
 5.0,
 5.0,
 4.333333333333333,
 5.0,
 4.333333333333333,
 5.0,
 4.333333333333333,
 5.0,
 3.6666666666666665]

# one way ANOVA test

In [14]:
# Shapiro-Wilk normality test
stats.shapiro(q1low_list), stats.shapiro(q1medium_list), stats.shapiro(q1large_list), stats.shapiro(q1xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς που δεν ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.9397180676460266, pvalue=0.1199113056063652),
 ShapiroResult(statistic=0.8506757616996765, pvalue=5.803517388880408e-14),
 ShapiroResult(statistic=0.7234978675842285, pvalue=7.276829987016307e-20),
 ShapiroResult(statistic=0.6508442163467407, pvalue=8.611391990598349e-07))

In [15]:
# Levene variance test  
stats.levene(q1low_list, q1medium_list, q1large_list, q1xlarge_list)

#Η τιμή p είναι μικρότερα από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με διαφορετική διακύμανση


LeveneResult(statistic=4.433478635808175, pvalue=0.004332324189802208)

In [16]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(q1low_list, q1medium_list, q1large_list, q1xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p)) 

F statistic = 12.022 and probability p = 0.000


# Kruskal Wallis test

In [17]:
stats.kruskal(q1low_list, q1medium_list, q1large_list, q1xlarge_list)

KruskalResult(statistic=32.92726688396111, pvalue=3.3364758125101153e-07)

# Dunn's test

In [18]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [q1low_list, q1medium_list, q1large_list, q1xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')


1         2         3         4
1  1.000000  0.004528  0.000016  0.000008
2  0.004528  1.000000  0.024527  0.012163
3  0.000016  0.024527  1.000000  0.440063
4  0.000008  0.012163  0.440063  1.000000

# Cohen's Effect Size

In [176]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(q1low_list), len(q1medium_list)
s1,s2= var(q1low_list, ddof=1), var(q1medium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1low_list), mean(q1medium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 0.8999279986847637


In [177]:
n1,n2= len(q1low_list), len(q1large_list)
s1,s2= var(q1low_list, ddof=1), var(q1large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1low_list), mean(q1large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 1.0228438132032183


In [178]:
n1,n2= len(q1low_list), len(q1xlarge_list)
s1,s2= var(q1low_list, ddof=1), var(q1xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1low_list), mean(q1xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 1.4092196809669186


In [179]:
n1,n2= len(q1medium_list), len(q1large_list)
s1,s2= var(q1medium_list, ddof=1), var(q1large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1medium_list), mean(q1large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.18671520327399113


In [180]:
n1,n2= len(q1medium_list), len(q1xlarge_list)
s1,s2= var(q1medium_list, ddof=1), var(q1xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1medium_list), mean(q1xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.5915258276495213


In [181]:
n1,n2= len(q1large_list), len(q1xlarge_list)
s1,s2= var(q1large_list, ddof=1), var(q1xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q1large_list), mean(q1xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.36273087792987024


# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.8999279986847637              medium-low --> Very Large: 0.8999279986847637
low-large --> Very Small:  -1.0228438132032183              large-low --> Very Large: 1.0228438132032183
low-xlarge --> Very Small: -1.4092196809669186              xlarge-low --> Very Large: 1.4092196809669186
medium-large --> Very Small: -0.18671520327399113           large-medium --> Medium: 0.18671520327399113
medium-xlarge --> Very Small: -0.5915258276495213           xlarge-medium --> Large: 0.5915258276495213 
large-xlarge --> Very Small: -0.36273087792987024           xlarge-large --> Medium: 0.36273087792987024

# CLUSTER(START) QUESTION 2
#Μού άρεσε που υπήρχε ο πράκτορας και έκανε παρεμβάσεις στη συζήτηση.

In [64]:
#Cluster Start 1 (LOW>4)
q2_1=dfquest2.groupby('Cluster (START)').get_group(1)


col=q2_1.columns
q2low=q2_1[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q2low.dropna(inplace=True)

meanq2low = q2low.mean(axis=0)#μεσος ορος μαθητων low στην ερώτηση 2


q2low_list=[]
for i, row in q2low.iterrows():
    q2low_list.append(q2low.at[i,'mean'])


q2low_list


<ipython-input-64-d2e92d76b319>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q2low.dropna(inplace=True)


[3.3333333333333335,
 3.6666666666666665,
 3.3333333333333335,
 1.0,
 3.3333333333333335,
 3.0,
 3.0,
 4.5,
 3.6666666666666665,
 2.3333333333333335,
 3.0,
 3.3333333333333335,
 2.3333333333333335,
 3.5,
 3.0,
 2.3333333333333335,
 2.6666666666666665,
 3.3333333333333335,
 4.666666666666667,
 2.6666666666666665,
 3.3333333333333335,
 3.3333333333333335,
 3.0,
 2.3333333333333335,
 2.0,
 3.6666666666666665,
 4.0]

In [65]:
#Cluster Start 2 (MEDIUM 4-5)
q2_2=dfquest2.groupby('Cluster (START)').get_group(2)


col=q2_2.columns
q2medium=q2_2[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q2medium.dropna(inplace=True)

meanq2medium = q2medium.mean(axis=0)#μεσος ορος μαθητων Medium στην ερώτηση 2


q2medium_list=[]
for i, row in q2medium.iterrows():
    q2medium_list.append(q2medium.at[i,'mean'])


q2medium_list


<ipython-input-65-d2afc62a4d81>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q2medium.dropna(inplace=True)


[4.0,
 3.6666666666666665,
 4.5,
 4.5,
 4.0,
 5.0,
 4.0,
 3.3333333333333335,
 3.0,
 5.0,
 2.0,
 4.0,
 4.0,
 4.0,
 2.0,
 3.0,
 3.0,
 4.666666666666667,
 3.6666666666666665,
 3.3333333333333335,
 3.0,
 4.5,
 4.0,
 3.0,
 3.0,
 1.0,
 2.0,
 5.0,
 4.0,
 4.333333333333333,
 5.0,
 5.0,
 1.0,
 3.0,
 3.0,
 4.0,
 4.666666666666667,
 2.6666666666666665,
 4.0,
 3.3333333333333335,
 4.333333333333333,
 4.0,
 3.6666666666666665,
 3.0,
 1.5,
 4.0,
 3.0,
 3.0,
 3.6666666666666665,
 4.0,
 3.6666666666666665,
 2.6666666666666665,
 3.0,
 5.0,
 3.6666666666666665,
 4.0,
 3.0,
 2.6666666666666665,
 4.666666666666667,
 4.333333333333333,
 3.0,
 3.3333333333333335,
 3.3333333333333335,
 3.0,
 4.333333333333333,
 4.0,
 1.5,
 4.0,
 5.0,
 2.6666666666666665,
 3.0,
 4.0,
 4.5,
 4.5,
 3.6666666666666665,
 4.0,
 4.0,
 3.0,
 3.6666666666666665,
 5.0,
 2.3333333333333335,
 5.0,
 2.6666666666666665,
 3.0,
 5.0,
 3.0,
 3.3333333333333335,
 3.6666666666666665,
 2.0,
 4.666666666666667,
 5.0,
 3.0,
 3.6666666666666665,


In [66]:
#Cluster Start 3 (LARGE 5-6)
q2_3=dfquest2.groupby('Cluster (START)').get_group(3)


col=q2_3.columns
q2large=q2_3[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q2large.dropna(inplace=True)

meanq2large = q2large.mean(axis=0)#μεσος ορος μαθητων Large στην ερώτηση 2


q2large_list=[]
for i, row in q2large.iterrows():
    q2large_list.append(q2large.at[i,'mean'])


q2large_list


<ipython-input-66-1287926099e4>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q2large.dropna(inplace=True)


[2.6666666666666665,
 5.0,
 4.0,
 5.0,
 4.0,
 3.6666666666666665,
 5.0,
 3.0,
 4.0,
 3.0,
 4.5,
 5.0,
 5.0,
 3.6666666666666665,
 3.0,
 4.0,
 4.0,
 5.0,
 4.666666666666667,
 2.6666666666666665,
 5.0,
 4.333333333333333,
 3.3333333333333335,
 4.0,
 4.666666666666667,
 4.0,
 4.333333333333333,
 3.6666666666666665,
 3.6666666666666665,
 3.0,
 4.5,
 4.0,
 4.5,
 4.5,
 3.0,
 1.0,
 3.6666666666666665,
 5.0,
 3.0,
 2.6666666666666665,
 3.6666666666666665,
 4.666666666666667,
 4.333333333333333,
 3.6666666666666665,
 3.3333333333333335,
 4.0,
 1.0,
 3.0,
 3.0,
 3.3333333333333335,
 4.333333333333333,
 2.5,
 3.3333333333333335,
 5.0,
 4.0,
 4.666666666666667,
 4.0,
 4.5,
 5.0,
 2.5,
 3.3333333333333335,
 3.3333333333333335,
 4.666666666666667,
 1.6666666666666667,
 4.333333333333333,
 5.0,
 4.0,
 3.6666666666666665,
 2.5,
 4.333333333333333,
 3.6666666666666665,
 3.0,
 5.0,
 5.0,
 5.0,
 2.0,
 4.0,
 2.3333333333333335,
 5.0,
 4.0,
 3.5,
 4.666666666666667,
 4.0,
 5.0,
 4.5,
 4.0,
 3.6666666666666

In [67]:
#Cluster Start 4 (XLARGE <6)
q2_4=dfquest2.groupby('Cluster (START)').get_group(4)


col=q2_4.columns
q2xlarge=q2_4[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q2xlarge.dropna(inplace=True)

meanq2xlarge = q2xlarge.mean(axis=0)#μεσος ορος μαθητων XLarge στην ερώτηση 2


q2xlarge_list=[]
for i, row in q2xlarge.iterrows():
    q2xlarge_list.append(q2xlarge.at[i,'mean'])


q2xlarge_list


<ipython-input-67-51fa5f142970>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q2xlarge.dropna(inplace=True)


[2.3333333333333335,
 4.333333333333333,
 2.0,
 4.0,
 2.6666666666666665,
 4.666666666666667,
 5.0,
 5.0,
 5.0,
 3.6666666666666665,
 4.0,
 5.0,
 5.0,
 5.0,
 4.0,
 4.0,
 3.6666666666666665,
 5.0,
 3.6666666666666665,
 4.333333333333333,
 4.333333333333333,
 5.0,
 5.0,
 4.0,
 3.3333333333333335,
 5.0,
 3.0]

# one way ANOVA test

In [68]:
stats.shapiro(q2low_list), stats.shapiro(q2medium_list), stats.shapiro(q2large_list), stats.shapiro(q2xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα δεν προέρχονται από πληθυσμούς που ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.9528084993362427, pvalue=0.2506225109100342),
 ShapiroResult(statistic=0.9549243450164795, pvalue=1.668240884100669e-06),
 ShapiroResult(statistic=0.9154202342033386, pvalue=1.5327472624449e-10),
 ShapiroResult(statistic=0.8623923659324646, pvalue=0.0020466025453060865))

In [69]:
# Levene variance test  
stats.levene(q2low_list, q2medium_list, q2large_list, q2xlarge_list)

#Η τιμή p είναι μεγαλύτερη από το όριο a = 0,05, 
#επομένως «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με την ίδια διακύμανση

LeveneResult(statistic=1.1349331598417542, pvalue=0.3343760340811747)

In [70]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(q2low_list, q2medium_list, q2large_list, q2xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p)) 

F statistic = 9.302 and probability p = 0.000


# Kruskal Wallis test

In [71]:
stats.kruskal(q2low_list, q2medium_list, q2large_list, q2xlarge_list)

KruskalResult(statistic=31.127937665956985, pvalue=7.989209807433341e-07)

# Dunn's test

In [72]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [q2low_list, q2medium_list, q2large_list, q2xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')


1         2         3         4
1  1.000000  0.020978  0.000058  0.000070
2  0.020978  1.000000  0.006285  0.019928
3  0.000058  0.006285  1.000000  0.873886
4  0.000070  0.019928  0.873886  1.000000

# Cohen's Effect Size

In [182]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(q2low_list), len(q2medium_list)
s1,s2= var(q2low_list, ddof=1), var(q2medium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2low_list), mean(q2medium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.5682672738196144


In [184]:
n1,n2= len(q2low_list), len(q2large_list)
s1,s2= var(q2low_list, ddof=1), var(q2large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2low_list), mean(q2large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 0.8563080742048942


In [185]:
n1,n2= len(q2low_list), len(q2xlarge_list)
s1,s2= var(q2low_list, ddof=1), var(q2xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2low_list), mean(q2xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 1.2674830510224124


In [186]:
n1,n2= len(q2medium_list), len(q2large_list)
s1,s2= var(q2medium_list, ddof=1), var(q2large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2medium_list), mean(q2large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.279952407119465


In [187]:
n1,n2= len(q2medium_list), len(q2xlarge_list)
s1,s2= var(q2medium_list, ddof=1), var(q2xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2medium_list), mean(q2xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.5685616690535787


In [188]:
n1,n2= len(q2large_list), len(q2xlarge_list)
s1,s2= var(q2large_list, ddof=1), var(q2xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q2large_list), mean(q2xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.2897677531941286


# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.5682672738196144               medium-low --> Large: 0.5682672738196144
low-large --> Very Small:  -0.8563080742048942                large-low -->  Very Large: 0.8563080742048942
low-xlarge --> Very Small: -1.2674830510224124               xlarge-low --> Very Large: 1.2674830510224124
medium-large --> Very Small: -0.279952407119465            large-medium --> Medium: 0.279952407119465
medium-xlarge --> Very Small: -0.5685616690535787             xlarge-medium --> Large: 0.5685616690535787 
large-xlarge --> Very Small: -0.2897677531941286            xlarge-large --> Medium: 0.2897677531941286

# CLUSTER(START) QUESTION 3
#Νομίζω ότι οι παρεμβάσεις του πράκτορα ήταν σχετικές με το τι συζητούσαμε εκείνη τη στιγμή.

In [74]:
#Cluster Start 1 (LOW>4)
q3_1=dfquest3.groupby('Cluster (START)').get_group(1)


col=q3_1.columns
q3low=q3_1[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q3low.dropna(inplace=True)

meanq3low = q3low.mean(axis=0)#μεσος ορος μαθητων low στην ερώτηση 3


q3low_list=[]
for i, row in q3low.iterrows():
    q3low_list.append(q3low.at[i,'mean'])


q3low_list


<ipython-input-74-64ff46336401>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q3low.dropna(inplace=True)


[3.3333333333333335,
 4.333333333333333,
 3.3333333333333335,
 1.0,
 3.0,
 3.0,
 3.0,
 3.5,
 4.333333333333333,
 3.0,
 3.3333333333333335,
 4.666666666666667,
 2.6666666666666665,
 3.5,
 3.6666666666666665,
 3.0,
 2.6666666666666665,
 3.0,
 4.666666666666667,
 3.0,
 5.0,
 3.0,
 2.0,
 2.6666666666666665,
 4.0,
 3.6666666666666665,
 4.0]

In [75]:
#Cluster Start 2 (MEDIUM 4-5)
q3_2=dfquest3.groupby('Cluster (START)').get_group(2)


col=q3_2.columns
q3medium=q3_2[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q3medium.dropna(inplace=True)

meanq3medium = q3medium.mean(axis=0)#μεσος ορος μαθητων Medium στην ερώτηση 3


q3medium_list=[]
for i, row in q3medium.iterrows():
    q3medium_list.append(q3medium.at[i,'mean'])


q3medium_list


<ipython-input-75-7bcd01d3a7a7>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q3medium.dropna(inplace=True)


[5.0,
 4.0,
 4.5,
 4.5,
 4.0,
 5.0,
 4.666666666666667,
 3.6666666666666665,
 5.0,
 4.0,
 2.0,
 4.0,
 4.0,
 3.6666666666666665,
 2.6666666666666665,
 4.0,
 2.3333333333333335,
 4.333333333333333,
 3.6666666666666665,
 3.3333333333333335,
 3.3333333333333335,
 5.0,
 3.6666666666666665,
 3.0,
 3.0,
 2.6666666666666665,
 4.0,
 4.0,
 4.5,
 4.0,
 3.6666666666666665,
 5.0,
 1.6666666666666667,
 3.0,
 5.0,
 4.5,
 4.666666666666667,
 3.6666666666666665,
 3.6666666666666665,
 3.6666666666666665,
 4.0,
 5.0,
 4.333333333333333,
 2.0,
 3.5,
 4.333333333333333,
 3.0,
 4.0,
 5.0,
 4.0,
 3.3333333333333335,
 2.6666666666666665,
 3.0,
 5.0,
 4.0,
 4.666666666666667,
 3.3333333333333335,
 3.0,
 3.6666666666666665,
 4.333333333333333,
 3.5,
 3.6666666666666665,
 3.3333333333333335,
 2.6666666666666665,
 5.0,
 4.0,
 2.0,
 4.0,
 5.0,
 3.0,
 3.3333333333333335,
 4.0,
 3.5,
 5.0,
 4.333333333333333,
 4.0,
 5.0,
 3.0,
 3.3333333333333335,
 5.0,
 2.6666666666666665,
 5.0,
 3.6666666666666665,
 4.666666666666

In [76]:
#Cluster Start 3 (LARGE 5-6)
q3_3=dfquest3.groupby('Cluster (START)').get_group(3)


col=q3_3.columns
q3large=q3_3[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q3large.dropna(inplace=True)

meanq3large = q3large.mean(axis=0)#μεσος ορος μαθητων Large στην ερώτηση 3


q3large_list=[]
for i, row in q3large.iterrows():
    q3large_list.append(q3large.at[i,'mean'])


q3large_list


<ipython-input-76-40370973e36d>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q3large.dropna(inplace=True)


[3.0,
 5.0,
 4.333333333333333,
 5.0,
 4.0,
 4.0,
 5.0,
 4.0,
 4.333333333333333,
 5.0,
 4.5,
 5.0,
 5.0,
 5.0,
 2.0,
 4.0,
 3.6666666666666665,
 5.0,
 4.666666666666667,
 3.0,
 5.0,
 4.0,
 3.3333333333333335,
 5.0,
 5.0,
 3.6666666666666665,
 4.333333333333333,
 3.0,
 4.333333333333333,
 3.3333333333333335,
 3.5,
 4.333333333333333,
 5.0,
 5.0,
 3.0,
 1.0,
 3.3333333333333335,
 1.0,
 4.0,
 2.6666666666666665,
 4.0,
 3.6666666666666665,
 4.666666666666667,
 3.3333333333333335,
 2.0,
 4.333333333333333,
 1.0,
 3.0,
 3.3333333333333335,
 3.6666666666666665,
 4.333333333333333,
 3.5,
 3.0,
 4.666666666666667,
 4.5,
 4.333333333333333,
 5.0,
 5.0,
 4.0,
 3.5,
 4.333333333333333,
 3.3333333333333335,
 4.666666666666667,
 4.0,
 4.666666666666667,
 5.0,
 4.0,
 3.3333333333333335,
 2.5,
 4.333333333333333,
 3.3333333333333335,
 3.3333333333333335,
 5.0,
 5.0,
 4.333333333333333,
 2.0,
 4.333333333333333,
 3.3333333333333335,
 4.333333333333333,
 4.333333333333333,
 3.5,
 4.666666666666667,
 4.

In [77]:
#Cluster Start 4 (XLARGE <6)
q3_4=dfquest3.groupby('Cluster (START)').get_group(4)


col=q3_4.columns
q3xlarge=q3_4[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q3xlarge.dropna(inplace=True)

meanq3xlarge = q3xlarge.mean(axis=0)#μεσος ορος μαθητων XLarge στην ερώτηση 3


q3xlarge_list=[]
for i, row in q3xlarge.iterrows():
    q3xlarge_list.append(q3xlarge.at[i,'mean'])


q3xlarge_list


<ipython-input-77-e919a252ed94>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q3xlarge.dropna(inplace=True)


[2.3333333333333335,
 5.0,
 2.0,
 4.0,
 2.6666666666666665,
 5.0,
 5.0,
 5.0,
 5.0,
 4.333333333333333,
 4.0,
 5.0,
 4.333333333333333,
 5.0,
 4.0,
 4.0,
 4.333333333333333,
 4.666666666666667,
 4.333333333333333,
 4.666666666666667,
 4.0,
 4.333333333333333,
 5.0,
 4.0,
 3.6666666666666665,
 5.0,
 4.0]

# one way ANOVA test

In [78]:
stats.shapiro(q3low_list), stats.shapiro(q3medium_list), stats.shapiro(q3large_list), stats.shapiro(q3xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς που δεν ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.9451513290405273, pvalue=0.16319632530212402),
 ShapiroResult(statistic=0.9495294690132141, pvalue=4.540277984688146e-07),
 ShapiroResult(statistic=0.888198971748352, pvalue=1.907958166458279e-12),
 ShapiroResult(statistic=0.8097025752067566, pvalue=0.00020249404769856483))

In [79]:
# Levene variance test  
stats.levene(q3low_list, q3medium_list, q3large_list, q3xlarge_list)

#Η τιμή p είναι μεγαλύτερη από το όριο a = 0,05, 
#επομένως «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με την ίδια διακύμανση

LeveneResult(statistic=0.48685798275436487, pvalue=0.6915497176604328)

In [80]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(q3low_list, q3medium_list, q3large_list, q3xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p)) 

F statistic = 8.485 and probability p = 0.000


# Kruskal Wallis test

In [81]:
stats.kruskal(q3low_list, q3medium_list, q3large_list, q3xlarge_list)

KruskalResult(statistic=30.428343864498277, pvalue=1.121439598101008e-06)

# Dunn's test

In [82]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [q3low_list, q3medium_list, q3large_list, q3xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')


1         2         3         4
1  1.000000  0.109616  0.000282  0.000415
2  0.109616  1.000000  0.001149  0.018604
3  0.000282  0.001149  1.000000  1.000000
4  0.000415  0.018604  1.000000  1.000000

# Cohen's Effect Size

In [189]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(q3low_list), len(q3medium_list)
s1,s2= var(q3low_list, ddof=1), var(q3medium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3low_list), mean(q3medium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.501040397078939


In [190]:
n1,n2= len(q3low_list), len(q3large_list)
s1,s2= var(q3low_list, ddof=1), var(q3large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3low_list), mean(q3large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.764214995111606


In [191]:
n1,n2= len(q3low_list), len(q3xlarge_list)
s1,s2= var(q3low_list, ddof=1), var(q3xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3low_list), mean(q3xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 1.0771021900543882


In [192]:
n1,n2= len(q3medium_list), len(q3large_list)
s1,s2= var(q3medium_list, ddof=1), var(q3large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3medium_list), mean(q3large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.2959088462019834


In [193]:
n1,n2= len(q3medium_list), len(q3xlarge_list)
s1,s2= var(q3medium_list, ddof=1), var(q3xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3medium_list), mean(q3xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.558955714476819


In [194]:
n1,n2= len(q3large_list), len(q3xlarge_list)
s1,s2= var(q3large_list, ddof=1), var(q3xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q3large_list), mean(q3xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.23896796179176452


# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.501040397078939            medium-low --> Large: 0.501040397078939
low-large --> Very Small:  -0.764214995111606            large-low -->  Large: 0.764214995111606
low-xlarge --> Very Small: -1.0771021900543882           xlarge-low --> Very Large: 1.0771021900543882
medium-large --> Very Small: -0.2959088462019834         large-medium --> Medium: 0.2959088462019834
medium-xlarge --> Very Small: -0.558955714476819         xlarge-medium --> Large: 0.558955714476819 
large-xlarge --> Very Small: -0.23896796179176452        xlarge-large --> Medium: 0.23896796179176452

# CLUSTER(START) QUESTION 4
#Νομίζω ότι οι παρεμβάσεις του πράκτορα με έκαναν να σκεφτώ καλύτερα γι αυτό που μαθαίνω.

In [101]:
#Cluster Start 1 (LOW>4)
q4_1=dfquest4.groupby('Cluster (START)').get_group(1)


col=q4_1.columns
q4low=q4_1[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q4low.dropna(inplace=True)

meanq4low = q4low.mean(axis=0)#μεσος ορος μαθητων low στην ερώτηση 4


q4low_list=[]
for i, row in q4low.iterrows():
    q4low_list.append(q4low.at[i,'mean'])


q4low_list

<ipython-input-101-d633f226b72b>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q4low.dropna(inplace=True)


[3.0,
 4.333333333333333,
 2.6666666666666665,
 1.0,
 2.0,
 2.0,
 3.0,
 3.5,
 3.3333333333333335,
 3.0,
 3.6666666666666665,
 3.6666666666666665,
 2.3333333333333335,
 4.0,
 3.3333333333333335,
 3.3333333333333335,
 2.3333333333333335,
 3.0,
 4.333333333333333,
 2.6666666666666665,
 3.0,
 2.6666666666666665,
 3.0,
 2.3333333333333335,
 2.6666666666666665,
 5.0,
 4.0]

In [102]:
#Cluster Start 2 (MEDIUM 4-5)
q4_2=dfquest4.groupby('Cluster (START)').get_group(2)


col=q4_2.columns
q4medium=q4_2[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q4medium.dropna(inplace=True)

meanq4medium = q4medium.mean(axis=0)#μεσος ορος μαθητων Medium στην ερώτηση 4


q4medium_list=[]
for i, row in q4medium.iterrows():
    q4medium_list.append(q4medium.at[i,'mean'])


q4medium_list

<ipython-input-102-406852c4922f>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q4medium.dropna(inplace=True)


[5.0,
 4.0,
 4.5,
 5.0,
 4.0,
 5.0,
 4.666666666666667,
 3.3333333333333335,
 4.0,
 4.333333333333333,
 2.0,
 3.3333333333333335,
 4.0,
 3.0,
 2.6666666666666665,
 3.0,
 2.3333333333333335,
 4.0,
 3.3333333333333335,
 3.3333333333333335,
 2.0,
 4.0,
 4.0,
 4.0,
 1.5,
 3.0,
 3.0,
 4.0,
 4.5,
 4.0,
 5.0,
 5.0,
 1.0,
 4.0,
 3.6666666666666665,
 4.0,
 4.333333333333333,
 2.6666666666666665,
 3.6666666666666665,
 3.6666666666666665,
 3.6666666666666665,
 5.0,
 3.0,
 4.0,
 1.5,
 4.666666666666667,
 3.0,
 4.0,
 4.333333333333333,
 4.333333333333333,
 3.6666666666666665,
 3.0,
 4.0,
 4.666666666666667,
 4.333333333333333,
 4.0,
 3.6666666666666665,
 2.3333333333333335,
 5.0,
 4.333333333333333,
 2.5,
 2.6666666666666665,
 3.0,
 3.0,
 3.0,
 4.5,
 1.5,
 4.0,
 5.0,
 2.6666666666666665,
 2.0,
 3.5,
 4.0,
 4.5,
 3.6666666666666665,
 4.0,
 4.0,
 3.6666666666666665,
 4.333333333333333,
 5.0,
 3.0,
 4.666666666666667,
 2.0,
 4.0,
 5.0,
 3.3333333333333335,
 3.3333333333333335,
 3.3333333333333335,
 3.

In [103]:
#Cluster Start 3 (LARGE 5-6)
q4_3=dfquest4.groupby('Cluster (START)').get_group(3)


col=q4_3.columns
q4large=q4_3[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q4large.dropna(inplace=True)

meanq4large = q4large.mean(axis=0)#μεσος ορος μαθητων Large στην ερώτηση 4


q4large_list=[]
for i, row in q4large.iterrows():
    q4large_list.append(q4large.at[i,'mean'])


q4large_list

<ipython-input-103-67fb0ad8e32a>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q4large.dropna(inplace=True)


[2.6666666666666665,
 5.0,
 3.3333333333333335,
 5.0,
 3.6666666666666665,
 3.6666666666666665,
 5.0,
 3.0,
 4.0,
 3.0,
 4.5,
 5.0,
 5.0,
 4.0,
 3.0,
 3.3333333333333335,
 4.0,
 5.0,
 4.333333333333333,
 2.3333333333333335,
 5.0,
 4.0,
 3.3333333333333335,
 5.0,
 5.0,
 4.0,
 4.333333333333333,
 3.3333333333333335,
 4.333333333333333,
 2.3333333333333335,
 4.5,
 3.6666666666666665,
 5.0,
 5.0,
 1.5,
 1.0,
 3.0,
 3.0,
 4.0,
 2.6666666666666665,
 3.3333333333333335,
 4.666666666666667,
 4.0,
 3.6666666666666665,
 3.0,
 3.6666666666666665,
 1.0,
 3.0,
 3.0,
 3.0,
 4.0,
 3.0,
 3.6666666666666665,
 4.0,
 4.0,
 4.0,
 5.0,
 4.0,
 4.5,
 3.0,
 3.3333333333333335,
 3.0,
 5.0,
 2.3333333333333335,
 3.6666666666666665,
 5.0,
 4.0,
 3.0,
 3.0,
 4.333333333333333,
 3.3333333333333335,
 4.666666666666667,
 5.0,
 5.0,
 5.0,
 1.0,
 5.0,
 2.3333333333333335,
 5.0,
 4.0,
 3.5,
 4.333333333333333,
 4.0,
 5.0,
 4.5,
 4.333333333333333,
 3.6666666666666665,
 4.0,
 4.0,
 4.666666666666667,
 3.0,
 1.0,
 4.6666

In [104]:
#Cluster Start 4 (XLARGE <6)
q4_4=dfquest4.groupby('Cluster (START)').get_group(4)


col=q4_4.columns
q4xlarge=q4_4[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q4xlarge.dropna(inplace=True)

meanq4xlarge = q4xlarge.mean(axis=0)#μεσος ορος μαθητων XLarge στην ερώτηση 4


q4xlarge_list=[]
for i, row in q4xlarge.iterrows():
    q4xlarge_list.append(q4xlarge.at[i,'mean'])


q4xlarge_list

<ipython-input-104-55768d76996a>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q4xlarge.dropna(inplace=True)


[2.3333333333333335,
 3.6666666666666665,
 2.0,
 4.333333333333333,
 3.0,
 4.333333333333333,
 5.0,
 5.0,
 5.0,
 4.666666666666667,
 5.0,
 5.0,
 4.333333333333333,
 5.0,
 4.0,
 4.0,
 4.0,
 5.0,
 4.666666666666667,
 4.0,
 3.3333333333333335,
 4.666666666666667,
 5.0,
 4.0,
 2.3333333333333335,
 5.0,
 3.3333333333333335]

# one way ANOVA test

In [105]:
stats.shapiro(q4low_list), stats.shapiro(q4medium_list), stats.shapiro(q4large_list), stats.shapiro(q4xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς που δεν ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.9787442088127136, pvalue=0.8329377770423889),
 ShapiroResult(statistic=0.9566468000411987, pvalue=2.5728129458002513e-06),
 ShapiroResult(statistic=0.9113034009933472, pvalue=7.4755777956792e-11),
 ShapiroResult(statistic=0.8499969244003296, pvalue=0.0011516254162415862))

In [106]:
# Levene variance test  
stats.levene(q4low_list, q4medium_list, q4large_list, q4xlarge_list)

#Η τιμή p είναι μεγαλύτερη από το όριο a = 0,05, 
#επομένως «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με την ίδια διακύμανση

LeveneResult(statistic=0.34553857981739994, pvalue=0.7924019788526453)

In [107]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(q4low_list, q4medium_list, q4large_list, q4xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p)) 

F statistic = 9.602 and probability p = 0.000


# Kruskal Wallis test

In [108]:
stats.kruskal(q4low_list, q4medium_list, q4large_list, q4xlarge_list)

KruskalResult(statistic=31.13748513071287, pvalue=7.952309957111306e-07)

# Dunn's test

In [109]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [q4low_list, q4medium_list, q4large_list, q4xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')


1         2         3         4
1  1.000000  0.022853  0.000066  0.000064
2  0.022853  1.000000  0.006457  0.016650
3  0.000066  0.006457  1.000000  0.780023
4  0.000064  0.016650  0.780023  1.000000

# Cohen's Effect Size

In [195]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(q4low_list), len(q4medium_list)
s1,s2= var(q4low_list, ddof=1), var(q4medium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4low_list), mean(q4medium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.5906807180933848


In [196]:
n1,n2= len(q4low_list), len(q4large_list)
s1,s2= var(q4low_list, ddof=1), var(q4large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4low_list), mean(q4large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 0.857156190467995


In [197]:
n1,n2= len(q4low_list), len(q4xlarge_list)
s1,s2= var(q4low_list, ddof=1), var(q4xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4low_list), mean(q4xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Very Large= 1.2136369001480025


In [198]:
n1,n2= len(q4medium_list), len(q4large_list)
s1,s2= var(q4medium_list, ddof=1), var(q4large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4medium_list), mean(q4large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.28671467115606614


In [199]:
n1,n2= len(q4medium_list), len(q4xlarge_list)
s1,s2= var(q4medium_list, ddof=1), var(q4xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4medium_list), mean(q4xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Large Effect Size= 0.581834773519262


In [200]:
n1,n2= len(q4large_list), len(q4xlarge_list)
s1,s2= var(q4large_list, ddof=1), var(q4xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q4large_list), mean(q4xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

Medium Effect Size= 0.28105891145849576


# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.5906807180933848               medium-low --> Large: 0.5906807180933848
low-large --> Very Small:  -0.857156190467995                large-low -->  Very Large: 0.857156190467995
low-xlarge --> Very Small: -1.2136369001480025               xlarge-low --> Very Large: 1.2136369001480025
medium-large --> Very Small: -0.28671467115606614            large-medium --> Medium: 0.28671467115606614
medium-xlarge --> Very Small: -0.581834773519262             xlarge-medium --> Large: 0.581834773519262 
large-xlarge --> Very Small: -0.28105891145849576            xlarge-large --> Medium: 0.28105891145849576

# CLUSTER(START) QUESTION 5
#Με ενόχλησαν οι παρεμβάσεις του πράκτορα καθώς συζητούσα με τον/την συνεργάτη μου.

In [7]:
#Cluster Start 1 (LOW>4)
q5_1=dfquest5.groupby('Cluster (START)').get_group(1)


col=q5_1.columns
q5low=q5_1[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q5low.dropna(inplace=True)

meanq5low = q5low.mean(axis=0)#μεσος ορος μαθητων low στην ερώτηση 5


q5low_list=[]
for i, row in q5low.iterrows():
    q5low_list.append(q5low.at[i,'mean'])


q5low_list

<ipython-input-7-ab727642d557>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q5low.dropna(inplace=True)


[1.6666666666666667,
 2.0,
 2.0,
 1.0,
 3.3333333333333335,
 1.0,
 1.0,
 2.5,
 1.6666666666666667,
 1.0,
 3.0,
 2.0,
 3.6666666666666665,
 1.0,
 3.0,
 2.0,
 3.6666666666666665,
 3.6666666666666665,
 1.0,
 3.3333333333333335,
 1.6666666666666667,
 1.6666666666666667,
 1.0,
 2.0,
 3.0,
 2.3333333333333335,
 2.3333333333333335]

In [8]:
#Cluster Start 2 (MEDIUM 4-5)
q5_2=dfquest5.groupby('Cluster (START)').get_group(2)


col=q5_2.columns
q5medium=q5_2[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q5medium.dropna(inplace=True)

meanq5medium = q5medium.mean(axis=0)#μεσος ορος μαθητων Medium στην ερώτηση 5


q5medium_list=[]
for i, row in q5medium.iterrows():
    q5medium_list.append(q5medium.at[i,'mean'])


q5medium_list

<ipython-input-8-df7ba05398e2>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q5medium.dropna(inplace=True)


[1.0,
 1.3333333333333333,
 1.5,
 5.0,
 4.0,
 1.0,
 1.0,
 1.0,
 3.0,
 1.0,
 2.0,
 2.6666666666666665,
 2.0,
 1.0,
 1.6666666666666667,
 1.0,
 1.6666666666666667,
 2.3333333333333335,
 4.0,
 4.666666666666667,
 1.0,
 1.0,
 1.6666666666666667,
 2.3333333333333335,
 5.0,
 3.6666666666666665,
 2.6666666666666665,
 1.0,
 2.5,
 1.6666666666666667,
 1.0,
 1.0,
 3.0,
 2.0,
 1.0,
 1.0,
 1.3333333333333333,
 2.3333333333333335,
 1.6666666666666667,
 3.0,
 3.0,
 1.0,
 3.0,
 2.0,
 4.0,
 1.6666666666666667,
 3.0,
 2.0,
 1.3333333333333333,
 1.3333333333333333,
 2.3333333333333335,
 1.6666666666666667,
 2.3333333333333335,
 1.0,
 1.6666666666666667,
 1.0,
 2.0,
 4.333333333333333,
 1.0,
 2.6666666666666665,
 1.5,
 3.6666666666666665,
 2.6666666666666665,
 1.0,
 1.3333333333333333,
 1.0,
 1.5,
 1.0,
 4.333333333333333,
 3.6666666666666665,
 2.0,
 3.5,
 4.0,
 2.0,
 1.6666666666666667,
 2.3333333333333335,
 3.0,
 3.6666666666666665,
 1.6666666666666667,
 3.6666666666666665,
 2.3333333333333335,
 2.3333

In [9]:
#Cluster Start 3 (LARGE 5-6)
q5_3=dfquest5.groupby('Cluster (START)').get_group(3)


col=q5_3.columns
q5large=q5_3[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q5large.dropna(inplace=True)

meanq5large = q5large.mean(axis=0)#μεσος ορος μαθητων Large στην ερώτηση 5


q5large_list=[]
for i, row in q5large.iterrows():
    q5large_list.append(q5large.at[i,'mean'])


q5large_list

<ipython-input-9-7bf16bad0025>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q5large.dropna(inplace=True)


[2.0,
 1.0,
 4.333333333333333,
 1.0,
 2.6666666666666665,
 3.0,
 1.0,
 3.5,
 1.6666666666666667,
 3.0,
 1.0,
 1.0,
 1.0,
 1.6666666666666667,
 2.0,
 2.6666666666666665,
 1.3333333333333333,
 1.0,
 1.0,
 2.3333333333333335,
 1.3333333333333333,
 1.3333333333333333,
 2.3333333333333335,
 4.0,
 1.0,
 3.6666666666666665,
 1.6666666666666667,
 2.0,
 3.0,
 2.3333333333333335,
 3.5,
 1.3333333333333333,
 1.0,
 1.0,
 3.0,
 1.0,
 3.6666666666666665,
 3.0,
 4.0,
 2.6666666666666665,
 2.0,
 1.0,
 1.6666666666666667,
 3.3333333333333335,
 2.0,
 1.0,
 5.0,
 1.3333333333333333,
 3.3333333333333335,
 1.0,
 3.0,
 3.5,
 3.3333333333333335,
 2.0,
 1.5,
 2.0,
 2.0,
 2.5,
 1.0,
 3.5,
 1.6666666666666667,
 1.3333333333333333,
 1.0,
 2.3333333333333335,
 3.0,
 1.0,
 4.0,
 1.3333333333333333,
 1.0,
 1.6666666666666667,
 2.3333333333333335,
 1.3333333333333333,
 1.5,
 1.0,
 1.0,
 4.5,
 2.3333333333333335,
 1.6666666666666667,
 1.0,
 2.0,
 3.0,
 1.3333333333333333,
 2.0,
 1.0,
 1.0,
 1.6666666666666667,
 2.0,

In [10]:
#Cluster Start 4 (XLARGE <6)
q5_4=dfquest5.groupby('Cluster (START)').get_group(4)


col=q5_4.columns
q5xlarge=q5_4[col[2:3:1]]#κρατάμε μόνο την στήλη των μέσων όρων
q5xlarge.dropna(inplace=True)

meanq5xlarge = q5xlarge.mean(axis=0)#μεσος ορος μαθητων XLarge στην ερώτηση 5


q5xlarge_list=[]
for i, row in q5xlarge.iterrows():
    q5xlarge_list.append(q5xlarge.at[i,'mean'])


q5xlarge_list

<ipython-input-10-09d66724ae15>:7: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  q5xlarge.dropna(inplace=True)


[3.6666666666666665,
 2.6666666666666665,
 2.0,
 1.6666666666666667,
 3.0,
 2.0,
 5.0,
 1.0,
 1.0,
 2.0,
 2.0,
 1.0,
 1.0,
 3.6666666666666665,
 2.0,
 2.6666666666666665,
 1.3333333333333333,
 2.3333333333333335,
 1.3333333333333333,
 1.0,
 2.0,
 1.0,
 1.0,
 1.3333333333333333,
 3.0,
 1.0,
 3.0]

# one way ANOVA test

In [11]:
stats.shapiro(q5low_list), stats.shapiro(q5medium_list), stats.shapiro(q5large_list), stats.shapiro(q5xlarge_list)

#Όλες οι τιμές p είναι μικρότερες από το όριο a = 0,05, 
#επομένως δεν «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς που δεν ακολουθούν την κανονική κατανομή

(ShapiroResult(statistic=0.8984958529472351, pvalue=0.012358708307147026),
 ShapiroResult(statistic=0.9113243222236633, pvalue=2.565785917063579e-10),
 ShapiroResult(statistic=0.8866769075393677, pvalue=1.526520338086812e-12),
 ShapiroResult(statistic=0.8729420900344849, pvalue=0.0033943031448870897))

In [12]:
# Levene variance test  
stats.levene(q5low_list, q5medium_list, q5large_list, q5xlarge_list)

#Η τιμή p είναι μεγαλύτερη από το όριο a = 0,05, 
#επομένως «αποτυγχάνουμε να απορρίψουμε» τη μηδενική υπόθεση 
#τα δείγματα προέρχονται από πληθυσμούς με την ίδια διακύμανση

LeveneResult(statistic=0.19150309210934993, pvalue=0.9021889746948921)

In [13]:
#Apply one way ANOVA by calling the f_oneway() method
F, p = stats.f_oneway(q5low_list, q5medium_list, q5large_list, q5xlarge_list)
print('F statistic = {:5.3f} and probability p = {:5.3f}'.format(F, p)) 

F statistic = 0.896 and probability p = 0.443


# Kruskal Wallis test

In [14]:
stats.kruskal(q5low_list, q5medium_list, q5large_list, q5xlarge_list)

KruskalResult(statistic=2.875187199344672, pvalue=0.41127184969127084)

# Dunn's test


In [15]:
#perform Dunn's test using a Bonferonni correction for the p-values

data = [q5low_list, q5medium_list, q5large_list, q5xlarge_list]
data

sp.posthoc_dunn(data , p_adjust = 'bonferroni')


1         2         3    4
1  1.0  1.000000  1.000000  1.0
2  1.0  1.000000  0.651944  1.0
3  1.0  0.651944  1.000000  1.0
4  1.0  1.000000  1.000000  1.0

# Cohen's Effect Size

In [1]:
from numpy import mean
from numpy import var
from math import sqrt

n1,n2= len(q5low_list), len(q5medium_list)
s1,s2= var(q5low_list, ddof=1), var(q5medium_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5low_list), mean(q5medium_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5low_list' is not defined

In [2]:
n1,n2= len(q5low_list), len(q5large_list)
s1,s2= var(q5low_list, ddof=1), var(q5large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5low_list), mean(q5large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5low_list' is not defined

In [3]:
n1,n2= len(q5low_list), len(q5xlarge_list)
s1,s2= var(q5low_list, ddof=1), var(q5xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5low_list), mean(q5xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5low_list' is not defined

In [4]:
n1,n2= len(q5medium_list), len(q5large_list)
s1,s2= var(q5medium_list, ddof=1), var(q5large_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5medium_list), mean(q5large_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5medium_list' is not defined

In [5]:
n1,n2= len(q5medium_list), len(q5xlarge_list)
s1,s2= var(q5medium_list, ddof=1), var(q5xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5medium_list), mean(q5xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5medium_list' is not defined

In [6]:
n1,n2= len(q5large_list), len(q5xlarge_list)
s1,s2= var(q5large_list, ddof=1), var(q5xlarge_list, ddof=1)
s= sqrt(((n1-1)*s1+(n2-1)*s2)/(n1+n2-2))
u1,u2= mean(q5large_list), mean(q5xlarge_list)
d= (u1-u2)/s

cohen= abs(d)
if cohen<0.10:
    print('Very Small Effect Size=',cohen)
elif cohen<0.1:
    print('Small Effect Size=',cohen)
elif cohen<0.50:
    print('Medium Effect Size=',cohen)
elif cohen<0.80:
    print('Large Effect Size=',cohen)
else:
    print('Very Large=', cohen)

NameError: name 'q5large_list' is not defined

# In the first directions of the pairs the results of effect size are <0 , in the opposite directions of the pairs the results of effect size are >0 :
low-medium --> Very Small: -0.0434049896143875            medium-low --> Very Small: 0.0434049896143875
low-large --> Medium:  0.10512972336855426                large-low -->  Very Small: -0.10512972336855426
low-xlarge --> Medium: 0.10665933107352765                xlarge-low --> Very Small: -0.10665933107352765
medium-large --> Medium: 0.14639572139480114              large-medium --> Very Small: -0.14639572139480114
medium-xlarge --> Medium: 0.14434656397416937             xlarge-medium --> Very Small: -0.14434656397416937 
large-xlarge --> Very Small: 0.0005800480789659743        xlarge-large --> Very Small: -0.0005800480789659743